tvx/scripts/paych2.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+import (
+	"os"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	init_ "github.com/filecoin-project/specs-actors/actors/builtin/init"
+	"github.com/filecoin-project/specs-actors/actors/builtin/paych"
+	"github.com/filecoin-project/specs-actors/actors/crypto"
+	"github.com/filecoin-project/specs-actors/actors/runtime/exitcode"
+
+	. "github.com/filecoin-project/oni/tvx/builders"
+	"github.com/filecoin-project/oni/tvx/schema"
+)
+
+// TODO -filter
+
+var initialBal = abi.NewTokenAmount(200_000_000_000)
+var toSend = abi.NewTokenAmount(10_000)
+
+func main() {
+	happyPathCreate()
+	happyPathUpdate()
+	happyPathCollect()
+}
+
+func happyPathCreate() {
+	metadata := &schema.Metadata{ID: "paych-ctor-ok", Version: "1.00", Desc: "happy path constructor"}
+
+	v := MessageVector(metadata)
+	v.Messages.SetDefaults(GasLimit(1_000_000_000), GasPrice(1))
+
+	// Set up sender and receiver accounts.
+	var sender, receiver AddressHandle
+	v.Actors.Accounts(address.SECP256K1, initialBal, &sender, &receiver)
+	v.CommitPreconditions()
+
+	// Add the constructor message.
+	createMsg := v.Messages.CreatePaymentChannelActor(sender.Robust, receiver.Robust, Value(toSend))
+	v.CommitApplies()
+
+	expectedActorAddr := AddressHandle{
+		ID:     MustNewIDAddr(MustIDFromAddress(receiver.ID) + 1),
+		Robust: sender.NextActorAddress(0, 0),
+	}
+
+	// Verify init actor return.
+	var ret init_.ExecReturn
+	MustDeserialize(createMsg.Result.Return, &ret)
+	v.Assert.Equal(expectedActorAddr.Robust, ret.RobustAddress)
```

Comment:
Nitpick: I am not sure why the asserters are part of the test vector. I think these are ortogonal and don't belong together.

Reply:
To (a) enable us to contextualise the error message and (b) so that it has access to the builder's context, including the state tree, messages, etc.